shared/profile/user/weboftrust/index.tsx

```diff
@@ -1,98 +1,128 @@
   username: string
-  verificationType: WebOfTrustVerificationType
-  vouchedAt: number
+  webOfTrustAttestation: {
+    attestation: string
+    attestingUser: string
+    proofID: string
+    status: WotStatusType
+    verificationType: WebOfTrustVerificationType
+    vouchedAt: number
+  }
 }
 
-const entry = (props: Props, statusStatement: string) => (
-  <>
-    <Kb.Box2 direction="horizontal" fullWidth={true}>
-      <Kb.Box2 direction="vertical" style={styles.avatarContainer} centerChildren={true}>
-        <Kb.Avatar
-          size={96}
-          username={props.attestingUser}
-          style={styles.avatar}
-          showFollowingStatus={false}
-        />
-      </Kb.Box2>
-      <Kb.Box2 direction="vertical" fullWidth={true} style={styles.textContainer} centerChildren={false}>
-        <Kb.Text style={styles.attestationText} type="BodySmall">
-          {statusStatement}
-        </Kb.Text>
-        <Kb.Text style={styles.attestationText} type="Body">
-          {props.attestation}
-        </Kb.Text>
-        <Kb.Box2 direction="vertical" style={styles.signatureBox} centerChildren={false} fullWidth={true}>
-          <Kb.Box2 direction="horizontal" gap="xxtiny" centerChildren={true} style={styles.innerSignatureBox}>
-            <Kb.Icon color={Styles.globalColors.blue} type="iconfont-proof-good" sizeType="Small" />
-            <Kb.Text type="BodySmall">signed by </Kb.Text>
-            <Kb.ConnectedUsernames
-              type={Styles.isMobile ? 'BodySmallBold' : 'BodyBold'}
-              usernames={props.attestingUser}
-              colorBroken={true}
-              colorFollowing={true}
-              style={styles.username}
-            />
-            <Kb.Text type="BodySmall">{formatTimeRelativeToNow(props.vouchedAt)}</Kb.Text>
+const WebOfTrust = (props: Props) => {
+  const dispatch = Container.useDispatch()
+  const {username, webOfTrustAttestation} = props
+  const {attestation, attestingUser, proofID, vouchedAt, status} = webOfTrustAttestation
+  const userIsYou = Container.useSelector(state => username === state.config.username)
+  const voucherIsYou = Container.useSelector(state => attestingUser === state.config.username)
+  const onAccept =
+    userIsYou && status === WotStatusType.proposed
+      ? () => dispatch(UsersGen.createWotReact({reaction: WotReactionType.accept, voucher: attestingUser}))
+      : null
+  const _onReject =
+    userIsYou && status === WotStatusType.proposed
+      ? () => dispatch(UsersGen.createWotReact({reaction: WotReactionType.reject, voucher: attestingUser}))
+      : null
+  const _onRevoke = voucherIsYou
+    ? () => dispatch(UsersGen.createSubmitRevokeVouch({proofID, voucheeName: username}))
+    : null
+  let onDelete, onReject
+  if (voucherIsYou) {
+    onDelete = _onRevoke
+  } else if (status === WotStatusType.accepted && userIsYou) {
+    onDelete = _onReject
+  } else if (userIsYou && status === WotStatusType.proposed) {
+    onReject = _onReject
+  }
```

Comment:
@xgess I remember that we talked about this, but since you've moved to hooks, I think this is the appropriate time to bring it up again:

1. I believe we try to set things to `undefined` rather than `null`.
2. If a callback **could be** `undefined` or `null`, then forgetting to conditionally check for it before calling can lead to a runtime error. Obviously TypeScript will give you a compile time error for this, so long as it's typed, but in principle it's easy to accidentally set something to `any` and get this issue as a runtime error.
3. Consider doing "nothing" inside of the callback rather than set it so `undefined` or `null`. E.g.

```tsx
const canReject = !userIsYou || status !== WotStatusType.proposed
const onReject = () => {
    if () {
        return
    }
    dispatch(UsersGen.createWotReact({ reaction: WotReactionType.reject, voucher: attestingUser })
}

// Inside render
{(canAccept || canReject || canDelete) && (
     <Kb.Box2 direction="horizontal" fullWidth={true} centerChildren={false} style={styles.buttonBar}>
        <Kb.ButtonBar align="flex-start">
```

Reply:
@thebearjew done!

Reply:
🎉